app/cdap/components/Connections/Browser/GenericBrowser/index.tsx

```diff
@@ -69,6 +69,7 @@ export function GenericBrowser({ selectedConnection }) {
   const loc = useLocation();
   const queryParams = new URLSearchParams(loc.search);
   const pathFromUrl = queryParams.get('path') || '/';
+  const [currentConnection, setCurrentConnection] = useState(selectedConnection);
```

Comment:
can the selected connection prop change?

Reply:
For this code I would suggest renaming to initialConnection

Reply:
Ack. Changing the prop name.